sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/SnapshotImpl.java

```diff
@@ -293,7 +319,44 @@ public Mono<Snapshot> createResourceAsync() {
             .serviceClient()
             .getSnapshots()
             .createOrUpdateAsync(resourceGroupName(), name(), this.innerModel())
-            .map(innerToFluentMap(this));
+            .map(innerToFluentMap(this))
+            .flatMap(snapshot -> {
+                if (snapshot.creationMethod() == DiskCreateOption.COPY_START && copyStartOptions.waitForCompletion) {
+                    return waitCopyStartToFinish();
```

Comment:
I am not sure if we should always wait for it here.

It might be better to return same, but having another `waitCopyStartToFinish()` method in Snapshot to let user do this. That might be more flexible for user.

OK, I see you have this option in "WithWaitForCopyStartCompletion". Personally I am a bit more inclined to the method approach.

Reply:
I see. The extra method is necessary if user choose not to wait, and itself will make the wait option during creation obsolete. Will change to the method approach.

Reply:
Will provide two methods:
```java
/**
 * Wait for CopyStart completion indefinitely unless errors are encountered.
 */
void waitForCopyStartCompletion();

/**
 * Wait for CopyStart completion for a specified timeout. If the operation is not complete when
 * the timeout is reached, a {@link TimeoutException} will be thrown.
 * @param maxWaitTime max timeout to wait for completion
 * @throws TimeoutException if operation operation is not complete after maxWaitTime
 */
void waitForCopyStartCompletion(Duration maxWaitTime) throws TimeoutException;
```

Reply:
Looks good. Consider uses ManagementException on timeout (or should we return e.g. boolean instead of throws on timeout?).

Reply:
OK, boolean is better.

Reply:
Suppose `await` and `waitFor` are synonym?

Reply:
seems so, try search which is more commonly used in our sdk or this repo.

Reply:
We got `SyncPoller.waitForCompletion` but throws exception when timeout. There's `CountDownLatch.await()` which is not in our repo but return false when timeout, like the example await above. 

Seems `waitFor` throws exception and `await` return false when dealing with timeout? I'm not sure. I'll change it to `awaitCopyStartCompletion` then..

Reply:
changed